builder/googlecompute/config.go

```diff
@@ -125,6 +127,11 @@ func NewConfig(raws ...interface{}) (*Config, []string, error) {
 		errs = packer.MultiErrorAppend(
 			errs, errors.New("a zone must be specified"))
 	}
+	if c.Region == "" && len(c.Zone) > 2 {
```

Comment:
Can you add a test for this? You can provide an example template and check whether `c.Region` is what you expect afterward.